staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -149,12 +165,30 @@ func (s *Validator) validateExpressions(fldPath *field.Path, sts *schema.Structu
 				// append a more descriptive error message. This error can only occur when static type checking has
 				// been bypassed. int-or-string is typed as dynamic and so bypasses compiler type checking.
 				errs = append(errs, field.Invalid(fldPath, obj, fmt.Sprintf("'%v': call arguments did not match a supported operator, function or macro signature for rule: %v", err, ruleErrorString(rule))))
+			} else if strings.HasPrefix(err.Error(), "operation cancelled: actual cost limit exceeded") {
```

Comment:
no test is exercising this matches... I changed the string locally and all tests still passed. we either need this to be typed in some way, or we need test coverage ensuring this remains functional

Reply:
This error occurs when per expression call exceed `perCallLimit` which configured during compilation. We have to update CEL to take perCallLimit while evaluation before we could update here and test it properly.

Reply:
I have added test for PerCallLimit and add comments to state the added input param is for testing and callers should always refer to const PerCallLimit. Thanks

Reply:
hmm... this is still not exercised by tests... I can change this to `if strings.HasPrefix(err.Error(), "bogus")` and unit tests still pass

Reply:
Thanks for catching up. I have fixed by reset `found` in test and update the error type

Reply:
sorry to keep making the same comment, but it doesn't look like the tests ensure this error string check match... the tests still pass with this change, which means they're not checking for a "call cost exceeds limit for rule" error:

```diff
diff --git a/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go b/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go
index 4a466189ec3..d69396f2b8f 100644
--- a/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go
+++ b/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go
@@ -184,7 +184,7 @@ func (s *Validator) validateExpressions(fldPath *field.Path, sts *schema.Structu
 				// append a more descriptive error message. This error can only occur when static type checking has
 				// been bypassed. int-or-string is typed as dynamic and so bypasses compiler type checking.
 				errs = append(errs, field.Invalid(fldPath, obj, fmt.Sprintf("'%v': call arguments did not match a supported operator, function or macro signature for rule: %v", err, ruleErrorString(rule))))
-			} else if strings.HasPrefix(err.Error(), "operation cancelled: actual cost limit exceeded") {
+			} else if strings.HasPrefix(err.Error(), "bogus") {
 				errs = append(errs, field.Invalid(fldPath, obj, fmt.Sprintf("'%v': call cost exceeds limit for rule: %v", err, ruleErrorString(rule))))
 			} else {
 				// no such key: {key}, index out of bounds: {index}, integer overflow, division by zero, ...
```